src/plugins/jingle/utils.js

```diff
@@ -0,0 +1,137 @@
+import { converse, api, _converse } from '@converse/headless/core';
+import JingleCallModal from "./modal/jingle-incoming-call-modal.js";
+import { html } from 'lit';
+
+const { Strophe, sizzle, $msg } = converse.env;
+const u = converse.env.utils;
+
+/**
+ * This function merges the incoming attributes and the jingle propose attribute into one.
+ * It also determines the type of the media i.e. audio or video
+ * @param { XMLElement } stanza
+ * @param { Object } attrs
+ */
+export function parseJingleMessage(stanza, attrs) {
+    if (isAJingleMessage(stanza) === true) {
+        const jingle_propose_type = getJingleProposeType(stanza);
+        return {
+            ...attrs, ...{
+                'jingle_propose': jingle_propose_type,
+                'jingle_retraction_id': getJingleRetractionId(stanza),
+                'template_hook': 'getJingleTemplate',
+                'jingle_status': jingleStatus(stanza)
+            }
+        }
+    }
+}
+
+function isAJingleMessage(stanza) {
+    if (jingleStatus(stanza) === 'incoming_call' || jingleStatus(stanza) === 'retracted' || jingleStatus(stanza) === 'call_ended') {
+        return true;
+    }
+    else false;
+}
+
+function jingleStatus(stanza) {
+    const el_propose = sizzle(`propose[xmlns="${Strophe.NS.JINGLEMESSAGE}"]`, stanza).pop();
+    const el_retract = sizzle(`retract[xmlns="${Strophe.NS.JINGLEMESSAGE}"]`, stanza).pop();
+    const el_finish = sizzle(`finish[xmlns="${Strophe.NS.JINGLEMESSAGE}"]`, stanza).pop();
+    if (el_propose) {
+        return 'incoming_call';
+    }
+    else if (el_retract) {
+        return 'retracted';
+    }
+    else if (el_finish) {
+        return 'call_ended';
+    }
+}
+
+function getJingleProposeType(stanza){
+    const el = sizzle(`propose[xmlns="${Strophe.NS.JINGLEMESSAGE}"] > description`, stanza).pop();
+    return el?.getAttribute('media');
+}
+
+function getJingleRetractionId(stanza){
+    const el = sizzle(`retract[xmlns="${Strophe.NS.JINGLEMESSAGE}"]`, stanza).pop();
+    return el?.getAttribute('id');
+}
+
+export function getJingleTemplate(model) {
+    return html`<converse-jingle-message jid='${model.get('jingle_status')}'></converse-jingle-message>`;
```

Comment:
You're not passing in the `jid` of the contact here, you're passing in `model.get('jingle_status')`.

That's most likely why you're not getting the model in the component.